zappa/cli.py

```diff
@@ -598,6 +598,14 @@ def dispatch_command(self, command, stage):
         """
 
         self.api_stage = stage
+        try:
```

Comment:
two minor comments:
- `check_stage_name` throws ` ValueError("AWS requires stage name to match a-zA-Z0-9_")`. You catch it here in `dispatch_command` and raise the same one with slightly more explanation. Could `check_stage_name` throw the long version (with "xyz does not" at the end) and we can avoid catching-and-raising it here?
- suuuper small detail, ignore if you want :) we can run `check_stage_name` before assigning it to `self.api_stage`. Looks a bit cleaner to me to validate before and the using it in the method

Reply:
Hmm, I've got no idea what I was thinking re-raising that Error, you're absolutely correct. Agreed on the second point, too.

Reply:
Ah, looking at the diff I think I was just moving code around… Fixed now, and looks much tidier!